utils/src/main/java/com/cloud/utils/nio/NioConnection.java

```diff
@@ -95,8 +113,8 @@ public void start() throws NioConnectionException {
         try {
             init();
         } catch (final ConnectException e) {
-            logger.warn("Unable to connect to remote: is there a server running on port " + _port);
-            return;
+            logger.warn("Unable to connect to remote: is there a server running on port" + _port, e);
```

Comment:
```suggestion
            logger.warn("Unable to connect to remote: is there a server running on port {}?", _port, e);
```
Maybe even change this message to error.